common/djangoapps/student/models.py

```diff
@@ -1114,7 +1114,9 @@ def get_enrollment(cls, user, course_key):
         if user.is_anonymous():
             return None
         try:
-            return cls.objects.get(
+            return cls.objects.select_related(
+                'course',
+            ).get(
```

Comment:
This makes me nervous... how sure are we that there will be a course overview for this enrollment whenever this is called?

Having this not return an enrollment if there is no course overview created feels like it could create some flakiness.

Reply:
I mean, selecting the related course overview if it exists is safe. It won't blow up if the overview is missing (it'll just blow up if you try to later access the overview). Also, @nasthagiri has sworn up and down that there shouldn't ever be a time when a course overview doesn't exist.

Reply:
Sigh... apparently, `select_related` does in fact cause a `DoesNotExist` if the related object doesn't exist.